### DISCUSSION TITLE
What is the best framework for automation cross-platform testing for Capacitor apps

### DISCUSSION THREAD

**Discussion by alexander-moskalenko** [ionic-team/capacitor#5271] [Q&A]
Hey guys, I am fairly new to test automation. 
Could you advise which framework I should look into it for automation  e2e and UI cross-platform testing for Capacitor apps?
Maybe you can point me to some resources.

**Comment by mlynch** [comment]
Absolutely. We're about to announce one ourselves for Capacitor and Ionic apps based on Appium and WebdriverIO, which are the two solutions I'd recommend for Capacitor E2E testing as they support testing your native app.

**Reply by cormack00** [reply]
Hi,
The ionic-e2e link was working but last week it was no longer working, and I don't see anything on https://ionic.io/docs.
 Is this framework still in development or I should just use something like WebdriverIO?

**Reply by mlynch** [reply]
Hey @cormack00, we're making a few small changes to this approach and actually instead of publishing a library to start we're going to publish a reference app using Appium and WebdriverIO to start. Keep an eye on this repo (not ready to use just yet): https://github.com/ionic-team/ionic-e2e-example

**Reply by cormack00** [reply]
Okay, thanks for the info
